docs/topics/testing.rst

```diff
@@ -61,21 +65,59 @@ To construct it, pass it an application and a scope::
     from channels.testing import ApplicationCommunicator
     communicator = ApplicationCommunicator(MyConsumer, {"type": "http", ...})
 
-To send an event, call ``send_input``::
+send_input
+~~~~~~~~~~
+
+Call it to send an event to the application::
 
     await communicator.send_input({
         "type": "http.request",
         "body": b"chunk one \x01 chunk two",
     })
 
-To receive an event, call ``receive_output``::
+receive_output
+~~~~~~~~~~~~~~
+
+Call it to receive an event from the application::
 
     event = await communicator.receive_output(timeout=1)
     assert event["type"] == "http.response.start"
 
-To wait for an application to exit (you'll need to either do this or wait for
-it to send you output before you can see what it did using mocks or inspection),
-use ``wait``::
+.. _application_communicator-receive_nothing:
+
+receive_nothing
+~~~~~~~~~~~~~~~
+
+Call it to check that there is no event waiting to be received from the
+application::
+
+    assert await communicator.receive_nothing(timeout=0.1, interval=0.01) is False
+    # Receive the rest of the http request from above
+    event = await communicator.receive_output()
+    assert event["type"] == "http.response.body"
+    assert event.get("more_body") is True
+    event = await communicator.receive_output()
+    assert event["type"] == "http.response.body"
+    assert event.get("more_body") is None
+    # Check that there isn't another event
+    assert await communicator.receive_nothing() is True
+    # You could continue to send and receive events
+    # await communicator.send_input(...)
+
+The method has two optional parameters:
+
+* ``timeout`` is ``0.1`` by default to keep tests short when there is no event
+  to receive and should be more than enough time.
+* ``interval`` is the time between two checks for new events. (It will return after
+  a check if there is an event.) The default of ``0.01`` should work well. However,
+  if you want to customize it you might use a low value (``0.001``) if you expect an
+  event and a high value (equal to ``timeout``) if you don't.
```

Comment:
Mention interval is in seconds as well, and drop commentary about how well the default works - just say the default is `0.01` and let people who need to tweak it do so. We don't want to have most people worrying about this value.